fabric8:watch fails

### Description

I tried to run `fabric8:watch` but got this error:

`[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.40:watch (default-cli) on project test: Execution default-cli of goal io.fabric8:fabric8-maven-plugin:3.5.40:watch failed: No <dockerHost> given, no DOCKER_HOST environment variable, no read/writable '/var/run/docker.sock' or '//./pipe/docker_engine' and no external provider like Docker machine configured -> [Help 1]`

I ran on OpenShift mode and built docker images on OpenShift nodes. This works for `fabric8:build`, so I think another local Docker engine is unnecessary for `fabric8:watch`.

	<properties>
		<fabric8.mode>openshift</fabric8.mode>
		<fabric8.skipResourceValidation>true</fabric8.skipResourceValidation>
		<fabric8.debug.enabled>true</fabric8.debug.enabled>
		<fabric8.namespace>test</fabric8.namespace>
	</properties>

### Info
* f-m-p version : 3.5.40
* Maven version (`mvn -v`) : 3.3.9
* Kubernetes / OpenShift setup and version : OpenShift 3.9

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -34,6 +34,7 @@ After this we will switch probably to real [Semantic Versioning 2.0.0](http://se
 * Fix NullPointerException in ClusterConfiguration during fabric8:watch
 * Feature 1498: Allow users to define secrets from annotations 
 * Feature 1498: Allow users to define secrets from annotations
+* Fix 1522: Remove need to initialize DockerAccess when on Openshift
 * Fix 1517: update vmp groupid and vert.x version
 
 ### 4.0.0-M2 (2018-12-14)

--- a/plugin/src/main/java/io/fabric8/maven/plugin/mojo/develop/WatchMojo.java
+++ b/plugin/src/main/java/io/fabric8/maven/plugin/mojo/develop/WatchMojo.java
@@ -161,10 +161,15 @@ public class WatchMojo extends io.fabric8.maven.docker.WatchMojo {
             return;
         }
 
+        log = new AnsiLogger(getLog(), useColor, verbose, !settings.getInteractiveMode(), getLogPrefix());
         clusterAccess = new ClusterAccess(getClusterConfiguration());
         kubernetes = clusterAccess.createDefaultClient(log);
 
-        super.execute();
+        if(clusterAccess.resolveRuntimeMode(mode, log).equals(RuntimeMode.kubernetes)) {
+            super.execute();
+        } else {
+            executeInternal(null);
+        }
     }
 
     protected ClusterConfiguration getClusterConfiguration() {
@@ -179,7 +184,9 @@ public class WatchMojo extends io.fabric8.maven.docker.WatchMojo {
 
     @Override
     protected synchronized void executeInternal(ServiceHub hub) throws MojoExecutionException {
-        this.hub = hub;
+        if(hub != null) {
+            this.hub = hub;
+        }
 
         URL masterUrl = kubernetes.getMasterUrl();
         KubernetesResourceUtil.validateKubernetesMasterUrl(masterUrl);
@@ -209,8 +216,7 @@ public class WatchMojo extends io.fabric8.maven.docker.WatchMojo {
     public WatcherContext getWatcherContext() throws MojoExecutionException {
         try {
             BuildService.BuildContext buildContext = getBuildContext();
-            WatchService.WatchContext watchContext = getWatchContext(hub);
-
+            WatchService.WatchContext watchContext = hub != null ? getWatchContext(hub) : null;
 
             return new WatcherContext.Builder()
                     .serviceHub(hub)
@@ -222,7 +228,6 @@ public class WatchMojo extends io.fabric8.maven.docker.WatchMojo {
                     .oldPodLogger(createLogger("[[R]][OLD][[R]] "))
                     .mode(mode)
                     .project(project)
-
                     .useProjectClasspath(useProjectClasspath)
                     .clusterConfiguration(getClusterConfiguration())
                     .kubernetesClient(kubernetes)

